sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/SubstituteUnresolvedOrdinals.scala

```diff
@@ -27,11 +27,19 @@ import org.apache.spark.sql.types.IntegerType
  * Replaces ordinal in 'order by' or 'group by' with UnresolvedOrdinal expression.
  */
 object SubstituteUnresolvedOrdinals extends Rule[LogicalPlan] {
-  private def isIntLiteral(e: Expression) = e match {
+  private def isIntLiteral(e: Expression): Boolean = e match {
     case Literal(_, IntegerType) => true
+    case Cube(groupByExprs) => groupByExprs.exists(isIntLiteral)
+    case Rollup(groupByExprs) => groupByExprs.exists(isIntLiteral)
```

Comment:
If you want to add `Cube` and `Rollup`, please change the method name. `isIntLiteral` matches what it does previously, but not now after this change.